src/analysis/normed_space/basic.lean

```diff
@@ -676,6 +676,35 @@ begin
   { exact smul_closed_ball' hc x r }
 end
 
+/-- A (semi) normed real vector space is homeomorphic to the unit ball in the same space.
+This homeomorphism sends `x : E` to `(1 + ∥x∥)⁻¹ • x`. -/
+def homeomorph_unit_ball {E : Type*} [semi_normed_group E] [semi_normed_space ℝ E] :
```

Comment:
Either this should have `@[simps]`, or you should write a custom lemma for it.

Reply:
I think about it as a statement "there exists a homeomorphism" and from this point of view I prefer to have it as a black box.

Reply:
If it's a black box then the conclusion should be `nonempty _` and it should be a lemma. But I don't think mathlib usually does that.

At any rate, it's not a black box because the docstring says what's in the box. Why not have a lemma that proves the docstring isn't lying?

Reply:
As a compromise, I've added non-simp `@[simps]` lemmas.

Reply:
I didn't know it was possible to do that in a single line, I assumed you had to do `attribute [-simp]` later. That's fine, to me the important thing is to have the lemma, not that it be simp